packages/react-dom/src/__tests__/ReactDOMSelect-test.js

```diff
@@ -808,4 +808,268 @@ describe('ReactDOMSelect', () => {
+          )),
+      ).toWarnDev('Invalid value for prop `value`');
+
+      expect(node.value).toBe('');
+    });
+
+    it('treats updated function value as an empty string', () => {
+      let node;
+
+      expect(
+        () =>
+          (node = ReactTestUtils.renderIntoDocument(
+            <select onChange={noop} value="monkey">
+              <option value={() => {}}>A function!</option>
+              <option value="monkey">A monkey!</option>
+              <option value="giraffe">A giraffe!</option>
+            </select>,
+          )),
+      ).toWarnDev('Invalid value for prop `value`');
+
+      expect(node.value).toBe('monkey');
+
+      node = ReactTestUtils.renderIntoDocument(
+        <select onChange={noop} value={() => {}}>
+          <option value={() => {}}>A function!</option>
+          <option value="monkey">A monkey!</option>
+          <option value="giraffe">A giraffe!</option>
+        </select>,
+      );
+
+      expect(node.value).toBe('');
+    });
+
+    it('treats updated function defaultValue as an empty string', () => {
+      let node;
+
+      expect(
+        () =>
+          (node = ReactTestUtils.renderIntoDocument(
+            <select defaultValue="monkey">
+              <option value={() => {}}>A function!</option>
+              <option value="monkey">A monkey!</option>
+              <option value="giraffe">A giraffe!</option>
+            </select>,
+          )),
+      ).toWarnDev('Invalid value for prop `value`');
+
+      expect(node.value).toBe('monkey');
+
+      node = ReactTestUtils.renderIntoDocument(
+        <select defaultValue={() => {}}>
+          <option value={() => {}}>A function!</option>
+          <option value="monkey">A monkey!</option>
+          <option value="giraffe">A giraffe!</option>
+        </select>,
+      );
+
+      expect(node.value).toBe('');
+    });
+
+    it('treats controlled function value as an empty string', () => {
+      let selectNode;
+
+      class Form extends React.Component {
+        state = {
+          value: 'giraffe',
+        };
+
+        handleChange = e => this.setState({value: e.target.value});
+
+        render() {
+          return (
+            <select
+              onChange={this.handleChange}
+              value={this.state.value}
+              ref={x => (selectNode = x)}>
+              <option value={() => {}}>A Function!</option>
+              <option value="monkey">A monkey!</option>
+              <option value="giraffe">A giraffe!</option>
+            </select>
+          );
+        }
+      }
+
+      expect(() => ReactTestUtils.renderIntoDocument(<Form />)).toWarnDev(
+        'Invalid value for prop `value`',
+      );
+
+      expect(selectNode.value).toBe('giraffe');
+
+      setUntrackedValue.call(selectNode, () => {});
+      dispatchEventOnNode(selectNode, 'change');
+
+      expect(selectNode.value).toBe('');
+    });
+
+    it('treats controlled function value as an empty string', () => {
```

Comment:
This seemed to be a additional case that I found failing before this PR. I don't know if this is the best way to test it, though.

Reply:
Actually, this case is probably fine as is, since the browser defaults to using the element's children as a value if the attribute `value` is not provided.

```
<select id="test">
  <option>hi</option>
</select>

console.log(document.getElementById('test').value) // 'hi'
```